tests/Tests/ORM/Functional/EagerFetchCollectionTest.php

```diff
@@ -96,6 +98,16 @@ public function testSubselectFetchJoinWithAllowedWhenOverriddenNotEager(): void
         $this->assertIsString($query->getSql());
     }
 
+    public function testSubselectFetchJoinWithAllowedWhenOverriddenNotEagerPaginator(): void
+    {
+        $query = $this->_em->createQuery('SELECT o, c FROM ' . EagerFetchOwner::class . ' o JOIN o.children c WITH c.id = 1');
+        $query->setMaxResults(1);
+        $query->setFetchMode(EagerFetchChild::class, 'owner', ORM\ClassMetadata::FETCH_LAZY);
+
+        $paginator = new Paginator($query, true);
+        $this->assertIsArray(iterator_to_array($paginator));
+    }
+
```

Comment:
This test doesn't look like that it is covering your changes. Your use case is that a query and dependencies of it gets cloned in the `LimitSubqueryOutputWalker`.

Reply:
Yup! This is a side effect, where paginating a query with `fetchMode` override didn't work.

I added a test case just for the code I modified.

Reply:
Thanks for adding a proper test for your use case.

I still don't understand the need for this test yet. Where does your code work with a fetchmode?

Reply:
If you look at previous test `testSubselectFetchJoinWithAllowedWhenOverriddenNotEager`, it asserts that fetch-mode override is currently honored.

This is the same but using `Paginator`. IMHO this is to assure that overriding fetch-mode is always working, using pagination or not.

But I agree that maybe it is redundant, in the sense that the error was not with fetch-mode or paginator directly.

If you prefer, I'll just won't include it.

Reply:
I found the time to checkout your branch and run the test locally to understand the need for this test and how the fetchmode affect it. The test helps to cover your changes :+1: